### ISSUE TITLE
Change the default value for micronaut.serde.serialization.inclusion to non_null

### ISSUE THREAD

**Issue by runenielsen** [micronaut-projects/micronaut-serialization#560]
### Feature description

The default for micronaut.serde.serialization.inclusion is `non_empty`. It should be changed to `non_null`.

Reason: It is not correct to just remove empty strings and empty arrays from the resulting JSON when serializing. This means that serialize + deserialize on some objects will result in objects, that are not a copy of the original.

`non_null` is a much more sensible default, that avoids this problem.

See also the failing test case testMicronautSerializationAndDeserializationWithEmptyProperties in SerializeEmptyPropertiesTest in https://github.com/runenielsen/serialize-empty-properties for a demonstration of a case where serialization + deserialization does not result in a copy of the original object.

All the failing test cases are fixed, when you add `micronaut.serde.serialization.inclusion=non_null` to `application.properties`.

It would also be nice, if the configuration possibilities for Micronaut Serialization was described in the documentation, so you don't have to look in the code for this 🙂 : https://micronaut-projects.github.io/micronaut-serialization/latest/guide/index.html

Thanks for a great framework! 🙂

**Comment by graemerocher** [comment]
would be a breaking change and if done can't be done until Micronaut 5

**Comment by graemerocher** [comment]
FWIW all configuration possibilities are described at https://micronaut-projects.github.io/micronaut-serialization/latest/guide/configurationreference.html

**Comment by runenielsen** [comment]
Ah, thanks. I'm blind and didn't notice that link, because it is in the page header. Sorry about that.

**Comment by distinctdan** [comment]
Yes, this shouldn't be the default. This isn't just a Java/Kotlin problem; missing arrays will also break javascript clients because it's not normal to have to check for the existence of an array. For most search API's, data will come back as an empty array if there aren't any results. It's also worth noting that as far as breaking changes go, this is pretty minor, because any properties that will come back could sometimes already be coming back.
